### PR TITLE
*: rewrite builtin func in to DNF

### PR DESCRIPTION
**Pull request by XuHuaiyu**
to #4080 
rewrite `a in (b, c, d)` to
`a = b or a = c or a = d`

PTAL @zz-jason @hanfei1991 @breeswish

### CODE REVIEW

**Comment by zz-jason** [comment]
@XuHuaiyu ci failed.

**Comment by hanfei1991** [comment]
This change will make great influence on refiner. Please comment the changed test instead of rewriting them and leave a TODO at same time.

File: plan/explain_test.go [hunk 1]
```diff
@@ -165,7 +165,7 @@ func (s *testExplainSuite) TestExplain(c *C) {
 			"select sum(t1.c1 in (select c1 from t2)) from t1",
```
Comment (zz-jason) [inline plan/explain_test.go]: add another two test like:
```sql
TiDB > desc select 1 in (select * from subq) from t;
+----------------+----------------+------------------------------+------+-------------------------------------------------+-------+
| id             | parents        | children                     | task | operator info                                   | count |
+----------------+----------------+------------------------------+------+-------------------------------------------------+-------+
| TableScan_8    |                |                              | cop  | table:t, range:(-inf,+inf), keep order:false    |     2 |
| TableReader_9  | HashSemiJoin_7 |                              | root | data:TableScan_8                                |     2 |
| TableScan_10   | Selection_11   |                              | cop  | table:subq, range:(-inf,+inf), keep order:false |   3.2 |
| Selection_11   |                | TableScan_10                 | cop  | eq(1, test.subq.a)                              |   3.2 |
| TableReader_12 | HashSemiJoin_7 |                              | root | data:Selection_11                               |   3.2 |
| HashSemiJoin_7 |                | TableReader_9,TableReader_12 | root | right:TableReader_12, aux                       |     2 |
+----------------+----------------+------------------------------+------+-------------------------------------------------+-------+

TiDB > desc select sum(6 in (select * from subq)) from t;
+----------------+----------------+-------------------------------+------+-------------------------------------------------+-------+
| id             | parents        | children                      | task | operator info                                   | count |
+----------------+----------------+-------------------------------+------+-------------------------------------------------+-------+
| TableScan_10   |                |                               | cop  | table:t, range:(-inf,+inf), keep order:false    |     4 |
| TableReader_11 | HashSemiJoin_9 |                               | root | data:TableScan_10                               |     4 |
| TableScan_12   | Selection_13   |                               | cop  | table:subq, range:(-inf,+inf), keep order:false |   3.2 |
| Selection_13   |                | TableScan_12                  | cop  | eq(6, test.subq.a)                              |   3.2 |
| TableReader_14 | HashSemiJoin_9 |                               | root | data:Selection_13                               |   3.2 |
| HashSemiJoin_9 | HashAgg_8      | TableReader_11,TableReader_14 | root | right:TableReader_14, aux                       |     4 |
| HashAgg_8      |                | HashSemiJoin_9                | root | type:complete, funcs:sum(join_5_aux_0)          |     1 |
+----------------+----------------+-------------------------------+------+-------------------------------------------------+-------+
```

**Comment by zz-jason** [comment]
@XuHuaiyu plz resolve conflicts.

**Pull request merged**